backend/data.go

```diff
@@ -37,11 +41,45 @@ func (fn QueryDataHandlerFunc) QueryData(ctx context.Context, req *QueryDataRequ
 // QueryDataRequest contains a single request which contains multiple queries.
 // It is the input type for a QueryData call.
 type QueryDataRequest struct {
+	ForwardHTTPHeaders
 	PluginContext PluginContext
 	Headers       map[string]string
```

Comment:
Nit: I was a bit confused here as how this would work. At first, I thought I could access the `Headers` directly to get http headers since the field is exported, but then I could not figure out how it would work, since the key is not canonicalized in `SetHTTPHeader`.
The correct way is using `GetHTTPHeader` or `GetHTTPHeaders`, however I feel like we need to specify that explicitly (that you must use `GetHTTPHeader` or `GetHTTPHeaders` to retrieve HTTP headers).
I'm not sure about this, maybe it's just me :thinking:  But I wanted to point it out anways

Reply:
Sorry, it's a way to say that this struct implements the `ForwardHTTPHeaders` interface. I removed this now and added a check that verifies the `QueryDataRequest` implements this interface instead.

Reply:
Much clearer now, thank you! :1st_place_medal: